### PR TITLE
Import latest firmware with Platform Abstraction Layer

### PR DESCRIPTION
**Pull request by arturkow2000**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+[package]
+name = "fobnail"
+version = "0.1.0"
+edition = "2021"
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[target.'cfg(target_os = "none")'.dependencies]
+pal_nrf = { path = "pal/pal_nrf" }
```
Comment (macpijan) [inline Cargo.toml:9]: @arturkow2000 Are you sure that the nrf dependencies are not pulled in for the PC build? I think I tried to build without nrf-hal repo and it failed (even when I build for PC target).

File: run-container.sh [hunk 2]
```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+dir=$(dirname $(readlink -f ${BASH_SOURCE[0]}))
+
+docker run --privileged \
```
Comment (macpijan) [inline run-container.sh:5]: @arturkow2000 the privileged flag is given twice - is it intended?

File: pal/pal_nrf/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+[dependencies]
+nrf52840-hal = { path = "../../external/nrf-hal/nrf52840-hal", features = ["rt"] }
+rtt-target = { version = "0.3.1", features = ["cortex-m"] }
+cortex-m = "0.7.3"
+cortex-m-rt = "0.7.0"
+usb-device = "0.2.8"
+# TODO: should use fobnail's repo or ideally should fetch from crates.io when
+# crate gets published
+usbd-ethernet = { git = "https://github.com/arturkow2000/usbd-ethernet", features = ["smoltcp_integration", "logging"] }
```
Comment (macpijan) [inline pal/pal_nrf/Cargo.toml]: @arturkow2000 Please use fobnail organization

File: Cargo.toml (hunk 1 above)
Comment (arturkow2000) [reply Cargo.toml:9]: @macpijan Seems like Cargo always parses metadata for all crates, even those not included by current target, probably nothing we can do here.

Build log reveals that `nrf-hal` is not built when building for PC.
```
   Compiling memchr v2.4.1
   Compiling libc v0.2.108
   Compiling log v0.4.14
   Compiling cfg-if v1.0.0
   Compiling quick-error v1.2.3
   Compiling regex-syntax v0.6.25
   Compiling termcolor v1.1.2
   Compiling managed v0.7.2
   Compiling byteorder v1.4.3
   Compiling bitflags v1.3.2
   Compiling humantime v1.3.0
   Compiling aho-corasick v0.7.18
   Compiling atty v0.2.14
   Compiling smoltcp v0.7.5
   Compiling regex v1.5.4
   Compiling env_logger v0.7.1
   Compiling pretty_env_logger v0.4.0
   Compiling pal_pc v0.1.0 (/hdd/projekty/fobnail/fobnail/pal/pal_pc)
   Compiling fobnail v0.1.0 (/hdd/projekty/fobnail/fobnail)
    Finished dev [unoptimized + debuginfo] target(s) in 9.85s
```

File: pal/pal_nrf/Cargo.toml (hunk 3 above)
Comment (arturkow2000) [reply pal/pal_nrf/Cargo.toml]: Done

File: run-container.sh (hunk 2 above)
Comment (arturkow2000) [reply run-container.sh:5]: Oversight, fixed.

**Review by macpijan — APPROVED** [review]

**Pull request merged**